gpu-deployment-tutorial/gpu-deployment-from-scratch.md

```diff
@@ -0,0 +1,270 @@
+# GPU Deployment Guide 
+Guide Adapted from:[**EuroSciPy 2025**](https://euroscipy.org/schedule/) | [Share Link](https://jacobtomlinson.dev/links/euroscipy-2025)
+
+## Agenda
+- Deployment (40 mins)
+  - NVIDIA Brev
+  - GPU Software Environment Fundamentals
+  - Python packages that use CUDA
+  - Monitoring/debugging tools
+  - Other platforms
+
+## Deployment
+
+The second half of the tutorial will discuss how to get your own GPUs on the cloud in more general terms. In order to dig into some of the things we will be learning we will be launching a VM through the [NVIDIA Brev](https://brev.nvidia.com) portal. 
+
+### Getting set up with Brev
+
+- Sign in to or register an account at https://brev.nvidia.com
+- Ensure you are a member of an organization
+  - One should be created for you when you register, but if not it will say "undefined" in the top right
+  - If you don't have one you can create a new one and give it a name
+- Apply credits to your organization
+  - Navigate to Billing
+  - Select "Redeem Code" <UPDATE IN HERE>
+
+<img width="1398" height="594" alt="Screenshot 2025-08-13 at 15 16 09" src="https://gist.github.com/user-attachments/assets/45f9b71a-eb5f-471f-93fa-2503cf8f4714" />
+
+#### Launching a Brev VM
+
+- Under "GPUs" select "New Instance"
+- Choose a GPU type that costs <$1/hour (e.g an L4) 
+- Choose any provider
+- Give your VM a name
+- Hit Deploy
+
+#### Connecting to your VM
+
+- Install the `brev` CLI
+  - `brew install brevdev/homebrew-brev/brev && brev login`
+- Connect via SSH
+  - `brev shell <your vm name>`
+
+For Linux and Windows instructions check the [brev-cli install documentation](https://docs.nvidia.com/brev/latest/brev-cli.html#installation-instructions)
+
+### Exploring our GPU Software Environment
+
+Let's start by exploring our VM to see what software we got out of the box.
+
+```console
+$ cat /etc/os-release
+PRETTY_NAME="Ubuntu 22.04.5 LTS"
+```
+
+We can check our GPU information by running `nvidia-smi`.
+
+```console
+$ nvidia-smi
```

Comment:
on an AWS instance I was not able to make it pass here. :/
 nvidia-smi errored. 
 ```
 NVIDIA-SMI has failed because it couldn't communicate with the NVIDIA driver. Make sure that the latest NVIDIA driver is installed and running.
 ```
 
 even though we had this in the instnace
 
 ```
 AMI Name: Deep Learning Base OSS Nvidia Driver GPU AMI (Ubuntu 22.04) 
Supported EC2 instances: G4dn, G5, G6, Gr6, G6e, P4d, P4de, P5, P5e, P5en, P6-B200
NVIDIA driver version: 580.95.05
CUDA versions available: cuda-12.6 cuda-12.8 cuda-12.9 cuda-13.0
Default CUDA version is 12.9
 ```